Assets/_Game/Scripts/Managers/States/StateManager.cs

```diff
@@ -19,14 +19,16 @@ public StateManager(GameSystems systems)
     /// <param name="stateType">State to push</param>
     /// <param name="popPreviousState">Option to pop current state off the stack</param>
     /// <param name="options">Optional data to be sent to the state</param>
-    public IEnumerator NavigateToState(Type stateType, bool popCurrentState = false, bool fadeTransition = false, Dictionary<string,object> options = null)
+    public void NavigateToState(Type stateType, bool popCurrentState = false, bool fadeTransition = false, Dictionary<string, object> options = null)
     {
+
+        GameSystems.Instance.StartChildCoroutine(NavigateToState_Implementation(stateType, popCurrentState = false, fadeTransition, options));
```

Comment:
Mostly there. We already have a cached ref for GameSystems so you don't need to grab the Instance. StartCoroutine can also just be called directly since GameSystems is already a Monobehavior. You don't need to make a separate method

```suggestion
        _systems.StartCoroutine(NavigateToState_Implementation(stateType, popCurrentState, fadeTransition, options));
```

Reply:
oy >_< That's a much easier way.